### PR TITLE
(Chore) Converts the ContainerSelect to ts

### PR DESCRIPTION
**Pull request by aditudorache**
This PR converts the last Component of the ContainerSelect to typescript

### CODE REVIEW

File: src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx [hunk 1]
```diff
@@ -1,33 +1,47 @@
...
-    selectedValue => {
+    (selectedValue: Item[] | null) => {
       parent.meta.updateIncident({ extra_container: selectedValue });
     },
     [parent]
   );
 
   const edit = useCallback(
-    event => {
+    (event: Event) => {
```
Comment (MauritsR) [inline src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx:37]: This is also a ClickEventHandler, like below

File: src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
 import Summary from './Summary';
-
-const ContainerSelect = ({ handler, meta, parent }) => {
-  const { value } = handler();
+import type { ClickEventHandler, Item, Meta } from './types';
+import type { Incident } from 'types/incident';
+import type { LatLngExpression } from 'leaflet';
+
+export interface ContainerSelectProps {
+  handler: () => { value: Item[] | null };
```
Comment (MauritsR) [inline src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx:12]: I see a lot of references to null, which in my opinion pollutes the code. Can we represent an empty selection with an empty array instead?

**Review by MauritsR — CHANGES_REQUESTED** [review]

File: src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx [hunk 3]
```diff
@@ -1,33 +1,47 @@
+import type { FunctionComponent } from 'react';
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx:1]: The trend is to refactor React import out as sinds 17.1 is not necessary (I believe not recommended) anymore  https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html. Also destructuring of types is in our code more the convention than the namespace notation. So I would prefer to keep up with the developments and move towards the `FunctionComponent` interface.

File: src/signals/incident/components/form/ContainerSelect/Selector/Selector.tsx [hunk 4]
```diff
@@ -100,7 +100,7 @@ const Selector = () => {
 
   const [, setMap] = useState();
 
-  const addContainer = useCallback<ClickEvent>(
+  const addContainer = useCallback<ClickEventHandler>(
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/Selector/Selector.tsx:103]: Will do!

File: src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx (hunk 2 above)
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx:12]: Agree. Changed that.

File: src/types/incident.ts [hunk 5]
```diff
@@ -29,14 +30,14 @@ export interface Datetime {
...
+export interface Location extends Record<string, any> {
   geometrie: Geometrie;
   address: Address;
 }
 
-export interface Address {
+export interface Address extends Record<string, any> {
   openbare_ruimte: string;
-  huisnummer: string;
+  huisnummer: string| number;
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/types/incident.ts:40]: There were some console errors here, I believe we send a string and receive a number from the api. Will have to figure out a solution in another PR

File: src/types/incident.ts [hunk 6]
```diff
@@ -29,14 +30,14 @@ export interface Datetime {
   info: string;
 }
 
-export interface Location {
+export interface Location extends Record<string, any> {
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/types/incident.ts:33]: The location json mock is more complete and we didn't made the full typing yet for this data structure.

File: src/signals/incident/components/form/ContainerSelect/types.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import type { IconOptions, LatLngExpression, Map } from 'leaflet';
 import type { ZoomLevel } from '@amsterdam/arm-core/lib/types';
 
-export type ClickEvent = (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
+export type ClickEventHandler = (event: React.SyntheticEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/types.ts:4]: SyntheticEvent is more general but in this case is not needed. I changed it back!

File: src/signals/incident/components/form/ContainerSelect/types.ts [hunk 8]
```diff
@@ -36,12 +36,12 @@ export interface Meta extends Record<string, unknown> {
 }
 
 export interface ContainerSelectValue {
-  selection: Item[];
+  selection: Item[] | null;
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/ContainerSelect/types.ts:39]: Sure! Why not!

File: src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (MauritsR) [reply src/signals/incident/components/form/ContainerSelect/ContainerSelect.tsx:1]: Cool, didn't know that! Better indeed to use FunctionComponent in this way.

**Review by MauritsR — APPROVED** [review]

**Pull request merged**